Fix flake8 errors

kajinamit opened the pull request:
Fix flake8 errors to improve readability of the code and also avoid potential bugs.

kajinamit committed:
Drop trailing semicolons

kajinamit committed:
Use print function, not print statement

print statement is no longer available in Python 3.

kajinamit committed:
Fix conflicting test case name

... and fix the wrong format of Sec-WebSocket-Protocol header in test
case.

kajinamit committed:
Drop unused/duplicate import

kajinamit committed:
tox: Restore pep8 target

... so that we can run flake8 check more easily. Note that the check
is not yet enforced in CI.

kajinamit committed:
Fix flake8 errors related to blank lines

kajinamit committed:
Fix flake8 errors related to whitespace

kajinamit committed:
Avoid multiple statements on one line

kajinamit committed:
Use 'is' in comparison to None

... instead of == or != .

kajinamit committed:
Remove dead code

... to fix "E265 block comment should start with '# '".

kajinamit committed:
Drop unused variables

kajinamit commented:
Note: Most of the fixes (especially fixes about blank line and white space errors) are generated by autopep8

CendioOssman reviewed (changes requested):
Nice cleanups!

Have you explored getting this in to GitHub actions as well? So we don't regress.

kajinamit committed:
Fix indentations

... which are now complained by flake8.

kajinamit committed:
Replace bare except

... to ensure the logic works only valid exceptions.

kajinamit committed:
Use explicit import

... to avoid pep8 errors caused by implicit import of a few modules.

kajinamit committed:
Add workflow to run pep8 check

kajinamit commented on websockify/websocketproxy.py:
```diff
@@ -293,20 +293,20 @@ class WebSocketProxy(websockifyserver.WebSockifyServer):
-        self.target_port    = kwargs.pop('target_port', None)
-        self.wrap_cmd       = kwargs.pop('wrap_cmd', None)
-        self.wrap_mode      = kwargs.pop('wrap_mode', None)
-        self.unix_target    = kwargs.pop('unix_target', None)
-        self.ssl_target     = kwargs.pop('ssl_target', None)
-        self.heartbeat      = kwargs.pop('heartbeat', None)
+        self.target_host = kwargs.pop('target_host', None)
+        self.target_port = kwargs.pop('target_port', None)
+        self.wrap_cmd = kwargs.pop('wrap_cmd', None)
+        self.wrap_mode = kwargs.pop('wrap_mode', None)
+        self.unix_target = kwargs.pop('unix_target', None)
+        self.ssl_target = kwargs.pop('ssl_target', None)
+        self.heartbeat = kwargs.pop('heartbeat', None)
```
The bare approach might be adding `noqa:` comment to each line. We can disable the check for the whole file but IMO that is not ideal.

kajinamit replied:
OK I think I could address this.

CendioOssman commented:
> I've added the workflow to run pep8 check. There are still number of checks disabled but at least we can detect any new errors. Does it make sense to you ?

Sounds great! It allows us to lock down what we already agree on, and explore other things as we want/need.

kajinamit committed:
Avoid nested import

... to get rid of F401 error by flake8.

kajinamit commented on tests/test_websockifyserver.py:
```diff
@@ -344,7 +344,6 @@ def test_fallback_sigchld_handler(self):
 
     def test_start_server_error(self):
         server = self._get_server(daemon=False, ssl_only=1, idle_timeout=1)
-        sock = server.socket('localhost')
```
Updated accordingly

kajinamit commented on websockify/token_plugins.py:
```diff
@@ -371,5 +371,5 @@ def lookup(self, token):
             return ['unix_socket', uds_path]
         except Exception as e:
-                logger.error("Error finding unix domain socket: %s" % str(e))
-                return None
+            logger.error("Error finding unix domain socket: %s" % str(e))
+            return None
```
Makes sense. Updated accordingly

kajinamit commented on tests/load.py:
```diff
@@ -43,8 +43,8 @@ def new_websocket_client(self):
-        cqueue = []
-        cpartial = ""
+        cqueue = []  # noqa: F841
+        cpartial = ""  # noqa: F841
```
Done

kajinamit commented on tests/test_token_plugins.py:
```diff
@@ -207,7 +207,7 @@ def test_asymmetric_jwe_token_plugin(self):
     def setUp(self):
         try:
-            import redis
+            import redis  # noqa: F401
```
I've moved this import to the top level.

kajinamit committed:
Fix flake8 caused by top level imports

kajinamit committed:
Ignore false flake8 error

kajinamit committed:
Extend max line length

The existing code does not expect 79 characters limit. Extend the limit
to avoid mass update.

kajinamit commented on tox.ini:
```diff
@@ -15,3 +15,7 @@ deps = -r{toxinidir}/test-requirements.txt
+
+[flake8]
+ignore = E126,E127,E128,E129,E401,E402,W504
+max-line-length = 160
```
OK I managed to get rid of all ignores.

kajinamit commented on websockify/websocketproxy.py:
```diff
@@ -298,15 +298,15 @@ def __init__(self, RequestHandlerClass=ProxyRequestHandler, *args, **kwargs):
-        self.host_token = kwargs.pop('host_token', None)
-        self.auth_plugin = kwargs.pop('auth_plugin', None)
+        self.host_token   = kwargs.pop('host_token', None)  # noqa: E221
+        self.auth_plugin  = kwargs.pop('auth_plugin', None)  # noqa: E221
```
It has been a while since we discussed this. I wonder if we can agree with following the pep8 guideline now (I mean, remove the adjustment commit for now) and "fix" these later if we want.
I know this is not urgent but I hope we can merge this soon, because this likely conflict with any new change, due to multiple lines being updated.

CendioOssman commented on websockify/websocketserver.py:
```diff
@@ -64,7 +64,7 @@ def _websocket_do_GET(self):
         self.do_GET = self._real_do_GET
 
         if (self.headers.get('upgrade') and
-            self.headers.get('upgrade').lower() == 'websocket'):
+                self.headers.get('upgrade').lower() == 'websocket'):
```
This change doesn't seem to be required by PEP8?

kajinamit replied:
No. This is required to address E125 Error

```
./websockify/websocketserver.py:67:13: E129 visually indented line with same indent as next logical line
```

CendioOssman replied:
This is controversial and not in alignment with PEP8:

https://github.com/PyCQA/pycodestyle/issues/126

So it apparently got split out to E129, rather than being grouped in with the okay things in E125.

But it's still enabled by default, which is also (understandably) controversial:

https://github.com/PyCQA/pycodestyle/issues/386

It would be nice if we could avoid having exceptions. But I don't want people fighting the linter either.

Since this is permitted in PEP8, I think the lesser evil is disabling E129 globally.

kajinamit replied:
OK. I disabled the check and reverted the change.

CendioOssman commented on websockify/sysloghandler.py:
```diff
@@ -110,5 +110,5 @@ def emit(self, record):
         except (KeyboardInterrupt, SystemExit):
             raise
-        except:
+        except Exception:
```
Looks like we can remove the special handling of `KeyboardInterrupt` and `SystemExit` with this change?

kajinamit replied:
Oh, yes. Because these two don't inherit Exception, these should not be caught here and are raised automatically.

CendioOssman reviewed (changes requested):
Thanks for all the adjustments! I think this is pretty much ready to go.

There was one indentation change that looked strange, though. Could you have a look at that, after which we can merge this.

kajinamit commented:
Thanks for the additional feedback !
The indentation change is what the E125 rule requires. Please see my inline comment.

CendioOssman commented on tox.ini:
```diff
@@ -18,3 +18,5 @@ deps = flake8
 [flake8]
 max-line-length = 160
+# E129 visually indented line with same indent as next logical line
+ignore = E121,E123,E126,E226,E24,E704,W503,W504,E129
```
Something went wrong here, this is not just E129?

kajinamit replied:
Adding ignore = E129 clears the checks disabled by default in flake8. I added E129 in addition to what are disabled by default.

kajinamit replied:
Updated the line to explicitly disable W504 (and W503 tightly coupled with W504), because the other checks are not failing. Maybe this is cleaner though it enables a few checks (eg. E121) which are disabled by default.

CendioOssman replied:
> Adding ignore = E129 clears the checks disabled by default in flake8. I added E129 in addition to what are disabled by default.

I see. What a confusing behaviour.

> Updated the line to explicitly disable W504 (and W503 tightly coupled with W504), because the other checks are not failing. Maybe this is cleaner though it enables a few checks (eg. E121) which are disabled by default.

Let's try it. We can disable them again later if we find them annoying. :)

CendioOssman commented on websockify/websocketserver.py:
```diff
@@ -64,7 +64,7 @@ def _websocket_do_GET(self):
 
         if (self.headers.get('upgrade') and
-                self.headers.get('upgrade').lower() == 'websocket'):
+            self.headers.get('upgrade').lower() == 'websocket'):
```
And please revert this change in the earlier commit, rather than changing it back and forth.

kajinamit replied:
Fixed accordingly.

kajinamit committed:
Ignore a few controversial errors

CendioOssman reviewed (approved):
Looks great!

Thanks for doing all the hard work here!
